testsuite/integration/manualmode/src/test/java/org/wildfly/test/manual/elytron/seccontext/ElytronClientDefaultSSLContextProviderTestCase.java

```diff
@@ -0,0 +1,95 @@
+import org.wildfly.security.x500.cert.BasicConstraintsExtension;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+import javax.net.ssl.SSLContext;
+import javax.security.auth.x500.X500Principal;
+import javax.ws.rs.client.ClientBuilder;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.Security;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+public class ElytronClientDefaultSSLContextProviderTestCase {
+
+    private static final String CONFIG_FILE = "file:./src/test/resources/wildfly-config-default-ssl-context.xml";
+    private static final String CLIENT_KEYSTORE = "client.keystore";
+    private static final char[] PASSWORD = "password".toCharArray();
+    private static final String ALIAS = "alias";
+    private static File KEYSTORES_DIR = new File("./target/keystores");
+
+    @BeforeClass
+    public static void generateKeystoreAndConfigureProvider() throws CertificateException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        new File(KEYSTORES_DIR, CLIENT_KEYSTORE).delete();
+        KEYSTORES_DIR.delete();
+        generateClientKeystore();
+        Security.insertProviderAt(new ElytronClientDefaultSSLContextProvider(CONFIG_FILE), 1);
+    }
+
+    @Test
+    public void testDefaultSSLContextIsNotReturnedIfNotConfigured() {
+        Assert.assertNotNull(Security.getProvider(ElytronClientDefaultSSLContextProvider.class.getSimpleName()));
+        try {
+            ResteasyClientBuilder builder = (ResteasyClientBuilder) ClientBuilder.newBuilder();
+            ResteasyClient resteasyClient = builder.sslContext(SSLContext.getDefault()).build();
+            Assert.assertEquals(resteasyClient.getSslContext().getProvider().getName(), ElytronClientDefaultSSLContextProvider.class.getSimpleName());
+        } catch (NoSuchAlgorithmException e) {
+            Assert.fail();
+        }
+    }
+
+    private static void generateClientKeystore() throws KeyStoreException, CertificateException, NoSuchAlgorithmException, IOException {
+        if (!KEYSTORES_DIR.exists()) {
+            KEYSTORES_DIR.mkdirs();
+        }
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        KeyStore clientKeyStore = KeyStore.getInstance("JKS");
+        clientKeyStore.load(null, null);
+
+        KeyPair clientKeyPair = keyPairGenerator.generateKeyPair();
+        PrivateKey signingKey = clientKeyPair.getPrivate();
+        PublicKey publicKey = clientKeyPair.getPublic();
+
+        X500Principal testClient10DN = new X500Principal("CN=" + ALIAS);
```

Comment:
Just for information, what does `testClient10DN` stand for?

Reply:
@fabiobrz That was just copied from other class, I changed this to be `issuerDN`. I also removed serialNumber and Extention from certificate as it was not necessary for this test to pass